src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy

```diff
@@ -325,9 +335,94 @@ class ModuleDistribution extends DefaultTask
                 }
             }
         }
+    }
+
+    // CONSIDER sort module by module dependencies
+    //   I don't think this is a problem with core labkey modules today, but it could be
+    private void warArchives()
+    {
+        StagingExtension staging = project.getExtensions().getByType(StagingExtension.class)
+        String warArchivePath = getWarArchivePath()
+
+        String[] modules = getModulesDir().list { File dir, String name -> name.endsWith(".module") }
+
+        // explode the modules before trying to assemble the .war
+        List<File> explodedModules = new ArrayList<>()
+        Arrays.asList(modules).forEach({ String moduleFile ->
+            String moduleName = moduleFile.substring(0, moduleFile.size() - ".module".size())
+            File explodedDir = new File(new File(warArchivePath).parentFile, moduleName)
+            explodedModules.push(explodedDir)
+            ant.unzip(src: new File(getModulesDir(),moduleFile), dest: explodedDir)
+        })
+
+        ant.zip(destfile: warArchivePath) {
+
+            zipfileset(dir: staging.webappDir,
+                    prefix: "") {
+                exclude(name: "WEB-INF/classes/distribution")
+            }
+
+            zipfileset(dir: staging.tomcatLibDir,
+                    prefix: "WEB-INF/lib/") {
+                exclude(name: "labkeyBootstrap.jar")
+                exclude(name: "bootstrap.jar")
+            }
+
+            // NOTE: if we want to enable running w/o LabKeyBootstrapClassLoader we have to do some of its work here (see ExplodedModule)
+            // NOTE: in particular some files need to be moved out of the module and into the WEB-INF directory
+
+            explodedModules.forEach({ File moduleDir ->
+                String moduleName = moduleDir.getName()
 
+                println("MODULE: ${moduleName}")
```

Comment:
This println should probably be removed.  If you want some echoing to go on, it's nicer to use `project.logger.debug` or `project.logger.quiet` (the equivalent of println), but without more info here, it's unclear who is telling you about a module.